addons/im_livechat/tests/test_chatbot_internals.py

```diff
@@ -108,3 +109,74 @@ def test_chatbot_welcome_steps(self):
         welcome_steps = self.chatbot_script._get_welcome_steps()
         self.assertEqual(len(welcome_steps), 1)
         self.assertEqual(welcome_steps, self.chatbot_script.script_step_ids[0])
+
+    def test_chatbot_multiple_rules_on_same_url(self):
+        bob_user = new_test_user(
+            self.env, login="bob_user", groups="im_livechat.im_livechat_group_user,base.group_user"
+        )
+        chatbot_no_operator, chatbot_operator = tuple(
+            self.env["chatbot.script"].create(
+                [
+                    {
+                        "title": "Chatbot operators not available",
+                        "script_step_ids": [
+                            Command.create(
+                                {
+                                    "step_type": "text",
+                                    "message": "I'm shown because there is no operator available",
+                                }
+                            )
+                        ],
+                    },
+                    {
+                        "title": "Chatbot operators available",
+                        "script_step_ids": [
+                            Command.create(
+                                {
+                                    "step_type": "text",
+                                    "message": "I'm shown because there is an operator available",
+                                }
+                            )
+                        ],
+                    },
+                ]
+            )
+        )
+        self.livechat_channel.user_ids += bob_user
+        self.livechat_channel.rule_ids = self.env["im_livechat.channel.rule"].create(
+            [
+                {
+                    "channel_id": self.livechat_channel.id,
+                    "chatbot_script_id": chatbot_no_operator.id,
+                    "chatbot_only_if_no_operator": True,
+                    "regex_url": "/",
+                    "sequence": 1,
+                },
+                {
+                    "channel_id": self.livechat_channel.id,
+                    "chatbot_script_id": chatbot_operator.id,
+                    "regex_url": "/",
+                    "sequence": 2,
+                },
+            ]
+        )
+        self.assertFalse(self.livechat_channel.available_operator_ids)
+        self.assertEqual(
+            self.env["im_livechat.channel.rule"]
+            .match_rule(self.livechat_channel.id, "/")
+            .chatbot_script_id,
+            chatbot_no_operator,
+        )
+        self.env["bus.presence"]._update_presence(
+            inactivity_period=0, identity_field="user_id", identity_value=bob_user.id
+        )
+        # FIXME: mail presence override is calling the wrong super method
+        bob_user.invalidate_recordset(["im_status"])
+        self.livechat_channel.invalidate_recordset(["available_operator_ids"])
```

Comment:
Clarify what's the functional intend of these 3 LOCs. It's far from obvious to call `_update_presence` manually.

I suppose this is to simulate operator becomes available, writing a brief comment like this above these LOCs would make it very clear what's the purpose here.